api/interservice/deployment/override_config.go

```diff
@@ -1,32 +1,63 @@
 package deployment
 
 import (
+	"github.com/sirupsen/logrus"
+
 	api "github.com/chef/automate/api/config/deployment"
 	config "github.com/chef/automate/api/config/shared"
 	w "github.com/chef/automate/api/config/shared/wrappers"
-	"github.com/sirupsen/logrus"
 )
 
-// NewUserOverrideConfigFromBackupRestoreTask takes a BackupRestoreTask and
-// converts the restoration configuration contained within into a new user
-// override AutomateConfig.
-func NewUserOverrideConfigFromBackupRestoreTask(req *BackupRestoreTask) *api.AutomateConfig {
-	// Convert a BackupRestoreTask into a sparse AutomateConfig that is suitable
-	// to be merged into the UserOverrideConfig
+// MergeAndValidateNewUserOverrideConfig takes an existing override config
+// and the BackupRestoreTask and builds/merges a new user override config.
+//
+// Validation of the configuration is a bit tricky. We still want to prevent
+// people from _adding_ deprecated values if the pass in set or patch config,
+// but we still want them to be able to restore if their config already has
+// deprecated values.
+func MergeAndValidateNewUserOverrideConfig(existing *api.AutomateConfig, req *BackupRestoreTask) error {
+	var err error
 	cfg := &api.AutomateConfig{}
+
+	// They gave us nothing, return the existing config.
 	if req == nil {
-		return cfg
+		return nil
 	}
-	if req.PatchConfig != nil {
-		cfg = req.PatchConfig
+
+	// If they passed a config set we'll only use that
+	if req.GetSetConfig() != nil {
```

Comment:
i feel like preventing people from setting deprecated config isn't right. As an example, you need to restore an old automate with the deprecated config to a new version, you probably still want the migrations to run